internal/configuration/configuration_test.go

```diff
@@ -0,0 +1,354 @@
+package configuration_test
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	gomock "github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/jakub-dzon/k4e-device-worker/internal/configuration"
+	"github.com/jakub-dzon/k4e-operator/models"
+)
+
+const (
+	deviceConfigName = "device-config.json"
+)
+
+var _ = Describe("Configuration", func() {
+
+	var (
+		datadir             string
+		err                 error
+		mockCtrl            *gomock.Controller
+		cfg                 models.DeviceConfigurationMessage
+		deviceConfiguration models.DeviceConfiguration
+	)
+
+	deviceConfigExists := func() {
+		_, err := os.Stat(fmt.Sprintf("%s/%s", datadir, deviceConfigName))
+		ExpectWithOffset(1, err).ToNot(HaveOccurred())
+	}
+
+	getDefaultDeviceconfig := func() models.DeviceConfiguration {
+		return models.DeviceConfiguration{
+			Heartbeat: &models.HeartbeatConfiguration{
+				HardwareProfile: &models.HardwareProfileConfiguration{
+					Include: false,
+					Scope:   "",
+				},
+				PeriodSeconds: 60,
+			},
+			Storage: nil,
+		}
+	}
+
+	BeforeEach(func() {
+		datadir, err = ioutil.TempDir("", "worloadTest")
+		Expect(err).ToNot(HaveOccurred())
+
+		mockCtrl = gomock.NewController(GinkgoT())
+
+		deviceConfiguration = models.DeviceConfiguration{
+			Heartbeat: &models.HeartbeatConfiguration{
+				PeriodSeconds: 1,
+			}}
+
+		cfg = models.DeviceConfigurationMessage{
+			Configuration:               &deviceConfiguration,
+			DeviceID:                    "",
+			Version:                     "",
+			Workloads:                   []*models.Workload{},
+			WorkloadsMonitoringInterval: 0,
+		}
+	})
+
+	AfterEach(func() {
+		mockCtrl.Finish()
+		_ = os.Remove(datadir)
+	})
+
+	Context("NewConfigurationManager", func() {
+		It("Raise an error if device-config file is not valid json", func() {
```

Comment:
The name suggests there should be some error raised, but the test does not check for any error. 
WDYT about something similar to `should use configuration when device-config file is not valid json`.